pkg/alertmanager/statefulset.go

```diff
@@ -240,11 +240,16 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config) (*appsv1.S
 
 	amArgs := []string{
 		fmt.Sprintf("--config.file=%s", alertmanagerConfFile),
-		fmt.Sprintf("--cluster.listen-address=[$(POD_IP)]:%d", 9094),
 		fmt.Sprintf("--storage.path=%s", alertmanagerStorageDir),
 		fmt.Sprintf("--data.retention=%s", a.Spec.Retention),
 	}
 
+	if *a.Spec.Replicas == 1 {
+		amArgs = append(amArgs, "--cluster.listen-address=")
```

Comment:
Can't we just skip the flag entirely?

Reply:
No, we have to provide empty value ([documentation](https://github.com/prometheus/alertmanager#turn-off-high-availability)). Address is 0.0.0.0:9094 by default ([code](https://github.com/prometheus/alertmanager/blob/master/cmd/alertmanager/main.go#L124))

Reply:
Thanks for the pointer. In that case, this lgtm.